app/controllers/two_factor_authentication/backup_code_verification_controller.rb

```diff
@@ -32,7 +32,10 @@ def all_codes_used?
 
     def handle_last_code
       BackupCodeGenerator.new(current_user).delete_existing_codes
-      redirect_to backup_code_depleted_url
+      generator = BackupCodeGenerator.new(current_user)
```

Comment:
Would it make sense to move this assignment up so that the `delete_existing_codes` can be called as `generator.delete_existing_codes` rather than on a separate instance?